Tests/VaporTests/MiddlewareTests.swift

```diff
@@ -29,4 +29,25 @@ final class MiddlewareTests: XCTestCase {
             XCTAssertEqual(res.body.string, "done")
         }
     }
+
+    func testCORSMiddleware() throws {
+        let app = Application(.testing)
+        defer { app.shutdown() }
+
+        app.grouped(
+            CORSMiddleware(configuration: .init(allowedOrigin: .originBased, allowedMethods: [.GET], allowedHeaders: [.origin]))
+        ).get("order") { req -> String in
+            return "done"
+        }
+
+        try app.testable().test(.GET, "/order", headers: ["Origin": "foo"]) { res in
```

Comment:
Small nit - this can be `app.test(...)` to clean it up a bit

```suggestion
        try app.test(.GET, "/order", headers: ["Origin": "foo"]) { res in
```